docs/root/intro/arch_overview/http_routing.rst

```diff
@@ -92,3 +94,27 @@ an upstream host. In the future Envoy will likely support true HTTP/2 priority o
 connection.
 
 The currently supported priorities are *default* and *high*.
+
+.. _arch_overview_http_routing_direct_response:
+
+Direct responses
+----------------
+
+Envoy supports the sending of "direct" responses. These are preconfigured HTTP responses
+that do not require proxying to an upstream server.
+
+There are two ways to specify a direct response in a Route:
+
+* Set the :ref:`direct_response <envoy_api_field_route.Route.direct_response>` field.
+  This works for all HTTP response statuses.
+* Set the :ref:`redirect <envoy_api_field_route.Route.redirect>` field. This works for
+  redirect response statuses only, but it simplifies the setting of the *Location* header.
+
+A direct response has an HTTP status code and an optional body. The Route configuration
+can specify the response body inline or specify the pathname of a file containing the
+body. If the Route configuration specifies a file pathname, Envoy will read the file
+upon configuration load and cache the contents. Regardless of whether the body is
+provided inline or in a file, the body length must be less than or equal to 4KB.
```

Comment:
nit: I might put the 4KB limitation inside an explicit `.. attention::` markup with a small explanation of why.